tests/user_test.py

```diff
@@ -234,6 +233,29 @@ def test_get_logs():
         db.get_logs( user_id=0, log_user_id = 0, security=security)
         db.get_logs( user_id=0, ipaddr = "", security=security)
 
+def test_course_list(new_user):
+    (user_email, api_key) = new_user
+    user_dict = db.validate_api_key(api_key)
+    user_id = user_dict['user_id']
+    primary_course_id = user_dict['primary_course_id']
+
+    recs1 = db.list_users(user_id=user_id)
+
+    matches = [rec for rec in recs1 if rec['user_id']==user_id]
+    assert(len(matches)>0)
+
+    # Make sure that there is an admin in the course who is not the user
+    recs2 = db.list_admins()
+    matches = [rec for rec in recs2 if rec['course_id']==primary_course_id and rec['user_id']!=user_id]
+    assert len(matches)==1
+
+    # Make sure that the endpoint works
+    with boddle(params={'api_key': api_key}):
+        res = bottle_app.api_list_users()
+    assert res['error']==False
+    assert res['result'] == recs1
```

Comment:
Ah, yes, further to my comment of db.py:359, notice that this test will pass even if db.list_users() returns an empty result set when it shouldn't. That is, this test doesn't test that bottle_app.api_list_users() and db.list_users() return correct results -- it only tests whether they both return the same result, regardless of the correctness of that result.

Reply:
OK, I committed a change to this test that makes sure that the length of the results are each 2. That should guard against at least returning an unexpected length, like I'm seeing in manual testing.